lib/common/validate.go

```diff
@@ -0,0 +1,113 @@
+package common
+
+import (
+	"reflect"
+	"regexp"
+
+	"gopkg.in/go-playground/validator.v8"
+	"fmt"
+)
+
+var validate *validator.Validate
+var nameRegex *regexp.Regexp
+var actionRegex *regexp.Regexp
+var protocolRegex *regexp.Regexp
+
+func init() {
+	// Initialise static data.
+	config := &validator.Config{TagName: "validate", FieldNameTag: "json"}
+	validate = validator.New(config)
+
+	var err error
+	if nameRegex, err = regexp.Compile("[a-zA-Z0-9_-]+"); err != nil { panic(err) }
```

Comment:
I've been doing `var fooRegex = regext.MustCompile(...)`  think that saves the err check